src/btree/bt_cursor.c

```diff
@@ -867,6 +870,42 @@ __cursor_equals(WT_CURSOR_BTREE *a, WT_CURSOR_BTREE *b)
 }
 
 /*
+ * __wt_btcur_compare_equal --
+ *	Return an equality comparison between two cursors.
+ */
+int
+__wt_btcur_compare_equal(
+    WT_CURSOR_BTREE *a_arg, WT_CURSOR_BTREE *b_arg, int *cmpp)
+{
+	WT_CURSOR *a, *b;
+	WT_SESSION_IMPL *session;
+
+	a = (WT_CURSOR *)a_arg;
+	b = (WT_CURSOR *)b_arg;
+	session = (WT_SESSION_IMPL *)a->session;
+
+	/* Confirm both cursors reference the same object. */
+	if (a_arg->btree != b_arg->btree)
+		WT_RET_MSG(
+		    session, EINVAL, "Cursors must reference the same object");
+
+	/*
+	 * The reason for a compare-equal method is because we can avoid doing
+	 * a full key comparison in some cases. If both cursors point into the
+	 * tree, take the fast path, otherwise fall back to the slower compare
+	 * method.
+	 */
+	if (F_ISSET(a, WT_CURSTD_KEY_INT) && F_ISSET(b, WT_CURSTD_KEY_INT))
+		*cmpp = !__cursor_equals(a_arg, b_arg);
+	else {
+		WT_RET(__wt_btcur_compare(a_arg, b_arg, cmpp));
+		if (*cmpp != 0)
+			*cmpp = 1;
+	}
+	return (0);
```

Comment:
I'm missing something: why do we ever need to take the slow path?  That is, if either cursor isn't positioned, they can't be equal by definition, right?

Or are you trying to cover a case where keys are set equal but the cursors are never used for anything?  I'd be more inclined to make it an error if either cursor isn't positioned...